### PR TITLE
352: unit tests

### PR DESCRIPTION
**Pull request by JasonGroven**
Unit tests added, mock-up bossy form added

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ module.exports = function(config) {
 			{ pattern: 'node_modules/@angular/**/*.js', included: false, watched: false },
 			{ pattern: 'node_modules/@angular/**/*.js.map', included: false, watched: false },
 
-			{ pattern: 'site/app/systemjs.config.js', included: false, watched: false },
+			{ pattern: 'site/app/systemjs.config.js', included: false, watched: true },
```
Comment (glawson) [inline karma.conf.js]: I'm pretty sure we don't want to watch this file.

Comment (tabersays) [inline karma.conf.js]: I would argue that we shouldn't watch any of the files as it is on single run and if you use npm test it should compile the ts to js before tests are run.

Comment (JasonGroven) [inline karma.conf.js]: I had singlerun set to false, and autowatch set to true, and I was testing things around with systemjs.config, then I set singlerun to true before pushing.  In hindsight, this file shouldn't've been included in the push.

**Comment by glawson** [comment]
resolves #352

**Pull request merged**